src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/contactperson/ContactPersonMappingApiServiceTest.kt

```diff
@@ -310,6 +311,47 @@ class ContactPersonMappingApiServiceTest {
     }
   }
 
+  @Nested
+  inner class GetByDpsContactAddressId {
+    @Test
+    internal fun `will pass oath2 token to service`() = runTest {
+      mockServer.stubGetByDpsContactAddressId(dpsContactAddressId = 1234567)
+
+      apiService.getByDpsContactAddressId(dpsContactAddressId = 1234567)
+
+      mockServer.verify(
+        getRequestedFor(anyUrl()).withHeader("Authorization", equalTo("Bearer ABCDE")),
+      )
+    }
+
+    @Test
+    internal fun `will pass DPS id to service`() = runTest {
+      mockServer.stubGetByDpsContactAddressId(dpsContactAddressId = 1234567)
+
+      apiService.getByDpsContactAddressId(dpsContactAddressId = 1234567)
+
+      mockServer.verify(
+        getRequestedFor(urlPathEqualTo("/mapping/contact-person/address/dps-contact-address-id/1234567")),
+      )
+    }
+
+    @Test
+    fun `will return dpsId`() = runTest {
+      mockServer.stubGetByDpsContactAddressId(
+        dpsContactAddressId = 1234567,
+        mapping = PersonAddressMappingDto(
+          dpsId = "1234567",
+          nomisId = 1234567,
+          mappingType = PersonAddressMappingDto.MappingType.MIGRATED,
+        ),
+      )
+
+      val mapping = apiService.getByDpsContactAddressId(dpsContactAddressId = 1234567)
+
+      assertThat(mapping.nomisId).isEqualTo(1234567)
```

Comment:
Might be better to have different numbers for dps / NOMIS - but meh

Reply:
true

Reply:
(though interestingly for person mapping they are the same)